lib/manageiq/api/common/headers.rb

```diff
@@ -0,0 +1,44 @@
+module HeaderScope
+  thread_mattr_accessor :current
+end
+
+module ManageIQ
+  module API
+    module Common
+      module Headers
+
+        def self.current=(val)
+          validate_headers(val)
+          HeaderScope.current = val
+        end
+
+        def self.current
+          HeaderScope.current
+        end
+
+        def self.decode(headers, key)
+          validate_headers(headers)
+          val = headers.instance_variable_get(:@req).stringify_keys
+          JSON.parse(Base64.decode64(val[key]))
```

Comment:
@syncrou Should there be a User class that can store the parsed properties and we could do 
User.current_user?

Reply:
@mkanoor - Yes, that will be set up as a separate PR.  I started with headers to make sure we were happy with the process etc.